var/spack/repos/builtin/packages/cantera/package.py

```diff
@@ -0,0 +1,197 @@
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
+
+
+class Cantera(Package):
+    """Cantera is a suite of object-oriented software tools for problems
+    involving chemical kinetics, thermodynamics, and/or transport processes."""
+
+    homepage = "http://www.cantera.org/docs/sphinx/html/index.html"
+    url      = "https://github.com/Cantera/cantera/archive/v2.2.1.tar.gz"
+
+    version('2.2.1', '9d1919bdef39ddec54485fc8a741a3aa')
+
+    variant('lapack',     default=True,  description='Build with external BLAS/LAPACK libraries')
+    variant('threadsafe', default=True,  description='Build threadsafe, requires Boost')
+    variant('sundials',   default=True,  description='Build with external Sundials')
+    variant('python',     default=False, description='Build the Cantera Python module')
+    variant('matlab',     default=False, description='Build the Cantera Matlab toolbox')
+
+    # Required dependencies
+    depends_on('scons')
+
+    # Recommended dependencies
+    depends_on('blas',      when='+lapack')
+    depends_on('lapack',    when='+lapack')
+    depends_on('boost',     when='+threadsafe')
+    depends_on('sundials',  when='+sundials')  # must be compiled with -fPIC
+
+    # Python module dependencies
+    extends('python', when='+python')
+    depends_on('py-numpy',  when='+python')
+    depends_on('py-scipy',  when='+python')
+    depends_on('py-cython', when='+python')
+    depends_on('py-3to2',   when='+python')
+    # TODO: these "when" specs don't actually work
+    # depends_on('py-unittest2',     when='+python^python@2.6')
+    # depends_on('py-unittest2py3k', when='+python^python@3.1')
+
+    # Matlab toolbox dependencies
+    # TODO: add Matlab package
+    # TODO: allow packages to extend multiple other packages
+    # extends('matlab',   when='+matlab')
+
+    def install(self, spec, prefix):
+        # Required options
+        options = [
+            'prefix={0}'.format(prefix),
+            'CC={0}'.format(os.environ['CC']),
+            'CXX={0}'.format(os.environ['CXX']),
+            'F77={0}'.format(os.environ['F77']),
+            'FORTRAN={0}'.format(os.environ['FC']),
+            'cc_flags=-fPIC',
+            # Allow Spack environment variables to propagate through to SCons
+            'env_vars=all'
+        ]
+
+        # BLAS/LAPACK support
+        if '+lapack' in spec:
+            options.extend([
+                'blas_lapack_libs=lapack,blas',
```

Comment:
This works for openblas. Does it work for other blas/lapack's? Another idea would be to take spec['lapack'].lapack_shared_lib, and strip off the path, extension, and lib.